### PR TITLE
feat(Menu): add Menu and MenuItem components

### PR DESCRIPTION
**Pull request by mnajdova**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/examples/fluent/FluentTheme.ts [hunk 1]
```diff
@@ -61,16 +61,34 @@ export const PlannerFluentTheme: IFluentThemeShape = {
...
+    FluentMenu: {
+      styles: () => ({
+        root: {
+          border: '1px solid red'
+        }
+      }),
+      variants: {
+        rounded: {
+          true: {
+            root: { borderRadius: '10px' }
```
Comment (mnajdova) [inline packages/office-ui-fabric-react/src/components/Button/examples/fluent/FluentTheme.ts:72]: - FluentMenu is not deifning styles for the FluentMenuItem, it just sends props (rounded, underlined..)
- FluentMenu defines the styles for the items too, but then it is problematic if you don't want to apply some of them on one item... (`<Menu item={[{rounded: true}, {rounded: false}]} />`)

File: packages/office-ui-fabric-react/src/components/Button/examples/fluent/compose.tsx [hunk 2]
```diff
@@ -93,3 +93,19 @@ export const compose = (displayName: string, BaseComponent: any) => (props: any)
 
   return <BaseComponent {...newProps} />;
 };
+
+export const composeNew = ({
+  themeName,
```
Comment (mnajdova) [inline packages/office-ui-fabric-react/src/components/Button/examples/fluent/compose.tsx]: displayName

File: packages/office-ui-fabric-react/src/components/Button/examples/fluent/FluentMenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { BaseMenuItem } from './../BaseMenuItem';
+import { compose } from './compose';
+// import { FluentMenu } from './';
+
+export const FluentMenuItem = compose(
+  'FluentMenuItem',
+  BaseMenuItem
+  // {
```
Comment (mnajdova) [inline packages/office-ui-fabric-react/src/components/Button/examples/fluent/FluentMenuItem.tsx:8]: We have to figure out circular slots definitions

File: packages/office-ui-fabric-react/src/components/Button/examples/fluent/compose.tsx [hunk 4]
```diff
@@ -57,23 +57,24 @@ export const getClassName = (
   const slotNames: string[] = Object.keys(componentStyles);
 
   if (theme && theme.components && theme.components[componentName] && theme.components[componentName].variants) {
-    Object.keys(theme.components[componentName].variants).forEach(variantName => {
-      stylesAdditions[variantName] = {};
-      variantNames.push(variantName);
-      Object.keys(theme.components[componentName].variants[variantName]).forEach(enumValue => {
-        const variant: any = {};
-        stylesAdditions[variantName][enumValue] = variant;
+    theme.components[componentName].variants.forEach((variants: any) => {
```
Comment (xugao) [inline packages/office-ui-fabric-react/src/components/Button/examples/fluent/compose.tsx]: whats this change about?

**Pull request merged**